### PR TITLE
Fix invisible avatars caused by connection resets

### PR DESCRIPTION
**Pull request by Atlante45**

### CODE REVIEW

**Comment by Atlante45** [comment]
# Test Plan:
1) Change your avatar a dozen times and make sure another avatar in the domain sees every change
2) Leave the domain and come back a dozen times and make sure someone in the domain sees you every time you come back.
3) Create and edit an entity and make sure someone in the domain sees your entity + edits.
4) Upload a model to the asset server and rez it in world. Make sure you can see the model load.

File: libraries/networking/src/udt/Connection.cpp [hunk 1]
```diff
@@ -60,6 +62,15 @@ Connection::Connection(Socket* parentSocket, HifiSockAddr destination, std::uniq
     _ack2Packet = ControlPacket::create(ControlPacket::ACK2, ACK2_PAYLOAD_BYTES);
     _lossReport = ControlPacket::create(ControlPacket::NAK, NAK_PACKET_PAYLOAD_BYTES);
     _handshakeACK = ControlPacket::create(ControlPacket::HandshakeACK, HANDSHAKE_ACK_PAYLOAD_BYTES);
+
+
+    // setup psuedo-random number generation for all instances of SendQueue
```
Comment (birarda) [inline libraries/networking/src/udt/Connection.cpp]: comment here is technically wrong now

File: libraries/networking/src/udt/SendQueue.h [hunk 2]
```diff
@@ -68,6 +68,8 @@ class SendQueue : public QObject {
     void setSyncInterval(int syncInterval) { _syncInterval = syncInterval; }
 
     void setProbePacketEnabled(bool enabled);
+
+    void setHandshakeRecieved() { _hasReceivedHandshakeACK = true; }
```
Comment (birarda) [inline libraries/networking/src/udt/SendQueue.h]: Did you mean to call this somewhere?

Comment (Atlante45) [reply libraries/networking/src/udt/SendQueue.h]: I did

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5615/HighFidelity-Beta-PR11979-58656f6655e5b556509839d20378cd5c8f00d783.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5620/HighFidelity-Beta-PR11979-3a45907df078bdc20bc915ac7103e810fbf28d5c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5620/HighFidelity-Beta-PR11979-3a45907df078bdc20bc915ac7103e810fbf28d5c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5627/HighFidelity-Beta-PR11979-44190f16c09fa3aae2330318c426364bbe0e8c47.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5627/HighFidelity-Beta-PR11979-44190f16c09fa3aae2330318c426364bbe0e8c47.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5635/HighFidelity-Beta-PR11979-ad547f483d76cb488f929dbc95874795e144e503.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5635/HighFidelity-Beta-PR11979-ad547f483d76cb488f929dbc95874795e144e503.dmg

**Comment by cain-kilgore** [comment]
Testing.

**Comment by cain-kilgore** [comment]
# Test Failed
Test | Result
--- | ---
Change your avatar a dozen times and make sure another avatar in the domain sees every change | :-1:
Leave the domain and come back a dozen times and make sure someone in the domain sees you every time you come back. | :-1:
Create and edit an entity and make sure someone in the domain sees your entity + edits. | 👍 
Upload a model to the asset server and rez it in world. Make sure you can see the model load. | 👍 

# Screenshots
## Models Appeared Okay
525f7d3f-a647-4f8e-847b-3f92206fe0ed

9e282411-a3dd-4e39-b019-ddc4573543bf

## Reconnecting to Domain Multiple Times resulted in the Avatar not appearing on around the 10th try. Second image shows that I am still able to create entities from the right side.
f11104d7-725e-4ff2-b060-eee3dc4f3692
205bffc1-998e-435a-b852-b385528cd4c8

## Avatar Change did not Happen, Animations Paused.
75237b3f-1d50-43fb-89d5-ba32035c987a

# ..
During this test I also had my client crash a couple times, I sent off the bugsplat. Here's the log files from both clients also attached.
[pc-1.zip](https://github.com/highfidelity/hifi/files/1558744/pc-1.zip) 
[pc-2.zip](https://github.com/highfidelity/hifi/files/1558750/pc-2.zip)

**Comment by birarda** [comment]
@cain-kilgore while testing, were both interface clients and the domain you were running in from this PR?

**Comment by cain-kilgore** [comment]
@birarda Yes

**Pull request closed without merging**